### PR TITLE
docs: Improve README.md

### PR DESCRIPTION
**Pull request by frankvonhoven**
## **Description**

Reduce README.md content to focus on building and running the app.

Documentation files linked on Table of Contents.

First iteration just focus on iOS platform.

## **Related issues**

Fixes: https://github.com/MetaMask/mobile-planning/issues/1533

## **Manual testing steps**

Comparisson

main: https://github.com/MetaMask/metamask-mobile/tree/main#readme
improved readme: https://github.com/MetaMask/metamask-mobile/tree/chore/tldr-readme#readme

### **Before**
Screenshot 2024-02-14 at 16 53 02

### **After**
Screenshot 2024-02-14 at 16 52 55

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [ ] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: scripts/build.sh [hunk 1]
```diff
@@ -198,20 +198,17 @@ buildAndroidRunFlask(){
...
-	SIM="${IOS_SIMULATOR:-"iPhone 13 Pro"}"
-	react-native run-ios --port=$WATCHER_PORT --simulator "$SIM"
+	react-native run-ios --port=$WATCHER_PORT
 }
 
 buildIosSimulatorQA(){
 	prebuild_ios
-	SIM="${IOS_SIMULATOR:-"iPhone 13 Pro"}"
-	react-native run-ios --port=$WATCHER_PORT --simulator "$SIM" --scheme "MetaMask-QA"
+	react-native run-ios --port=$WATCHER_PORT --scheme "MetaMask-QA"
```
Comment (Cal-L) [inline scripts/build.sh:206]: What is the reason for these changes?

**Review by Cal-L — COMMENTED** [review]
left a question

File: scripts/build.sh (hunk 1 above)
Comment (jpcloureiro) [reply scripts/build.sh:206]: This is a friction point to new contributors because iPhone 13 Pro isn't available by default on current Xcode version (15).

We can let react-native CLI default to iPhone 14 SIM.

File: README.md [hunk 2]
```diff
@@ -10,356 +10,59 @@ For up to the minute news, follow our [Twitter](https://twitter.com/metamask) or
...
-
-#### Android
-
-- Install [Java](https://www.java.com/en/download/). To check if Java is already installed, run:
-  ```
-  java -version
-  ```
-- Install the Android SDK, via [Android Studio](https://developer.android.com/studio).
-  - _MetaMask Only:_ To create production builds, you need to install Google Play Licensing Library via the SDK Manager in Android Studio.
-- Install the Android NDK (version `21.4.7075529`), via [Android Studio](https://developer.android.com/studio)'s SDK Manager.
```
Comment (sethkfman) [inline README.md:63]: I am pretty sure this should be 24. You can reference the bitrise.yml for the version just search NDK

Comment (jpcloureiro) [reply README.md:63]: We've [updated the NDK version here](https://github.com/MetaMask/metamask-mobile/pull/8492/commits/0b5eec78dc651e78d3be08760d62fdbf270dd19b) but seems we've missed another reference

I'll remove the duplicate.
Thanks @sethkfman

Comment (jpcloureiro) [reply README.md:63]: addressed [2e9c294](https://github.com/MetaMask/metamask-mobile/pull/8492/commits/2e9c2943714d8a8cd5e204f0e19da4f6db76b187)

**Review by sethkfman — APPROVED** [review]
LGTM

File: scripts/build.sh (hunk 1 above)
Comment (sethkfman) [reply scripts/build.sh]: Oh man! I have been dealing with issue for a while. Thx!
